### PR TITLE
User flow attributes doc creation

### PR DESCRIPTION
**Pull request by jkdouglas**
Docs for the userFlowAttributes API

### CODE REVIEW

File: api-reference/beta/api/userflowattribute-post-userflowattributes.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+|Authorization|Bearer {token}. Required.|
+|Content-Type|application/json. Required.|
+
+## Request body
+
+In the request body, provide a JSON representation of [userFlowAttribute](../resources/userflowattributes.md). All the properties listed in the following table are required.
+
+|Property|Type|Description|
+|:---------------|:--------|:----------|
+|displayName|String|The display name of the user flow attribute.|
```
Comment (agrabhi) [inline api-reference/beta/api/userflowattribute-post-userflowattributes.md:54]: id and userFlowAttributeType is missing

Comment (agrabhi) [reply api-reference/beta/api/userflowattribute-post-userflowattributes.md:54]: will help to run the sample request, response given here against prod

Comment (agrabhi) [reply api-reference/beta/api/userflowattribute-post-userflowattributes.md:54]: Location header?

File: api-reference/beta/api/userflowattributes-delete.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ms.prod: "microsoft-identity-platform"
+---
+
+# Delete userFlowAttribute
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Delete a [userFlowAttribute](../resources/userflowattributes.md).
```
Comment (agrabhi) [inline api-reference/beta/api/userflowattributes-delete.md:16]: we must mention that only user created (custom attributes) can be deleted.

File: api-reference/beta/api/userflowattributes-get.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "microsoft.graph.userFlowAttributes"
+} -->
+
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+
+{
```
Comment (agrabhi) [inline api-reference/beta/api/userflowattributes-get.md:85]: in all the examples @odata.type field is missing - should that be added?

File: api-reference/beta/api/userflowattributes-update.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+ms.prod: "microsoft-identity-platform"
+---
+
+# Update userFlowAttributes
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Update the properties of a [userFlowAttribute](../resources/userflowattributes.md) object.
```
Comment (agrabhi) [inline api-reference/beta/api/userflowattributes-update.md:16]: we should mention only custom attributes can be updated

File: api-reference/beta/resources/userflowattributes.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+# userFlowAttributes resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents user flow attributes in an Azure Active Directory tenant and an Azure AD B2C tenant.
+
+Configuring user flow attributes in your Azure Active Directory or Azure AD B2C tenant allows you collect information about a user during sign up. There is a built-in set of attributes that you can choose to collect, like Given Name, Surname, City, and Postal Code. You can also configure custom user flow attributes to collect information from a user that is not built-in to the directory. Custom user flow attributes are an abstraction over Azure Active Directory schema extensions.
```
Comment (agrabhi) [inline api-reference/beta/resources/userflowattributes.md:18]: we should link with `Azure Active Directory schema extensions`  documentation

File: api-reference/beta/resources/userflowattributes.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+## Methods
+
+| Method       | Return Type  |Description|
+|:---------------|:--------|:----------|
+|[List](../api/userflowattributes-list.md)|userFlowAttributes collection|Retrieve all built in and custom user flow attributes.|
+|[Create](../api/userflowattribute-post-userflowattributes.md)|userFlowAttribute|Create a new custom user flow attribute.|
+|[Get](../api/userflowattributes-get.md) |userFlowAttribute|Retrieve properties of a user flow attribute.|
+|[Update](../api/userflowattributes-update.md)|None|Update a custom user flow attribute.|
+|[Delete](../api/userflowattributes-delete.md)|None|Delete a custom user flow attribute.|
+
```
Comment (agrabhi) [inline api-reference/beta/resources/userflowattributes.md:29]: we can mention here as well in 'emphasis' that only custom attributes can be modified.

File: api-reference/beta/api/userflowattribute-post-userflowattributes.md (hunk 1 above)
Comment (jkdouglas) [reply api-reference/beta/api/userflowattribute-post-userflowattributes.md:54]: Fixed

File: api-reference/beta/api/userflowattributes-delete.md (hunk 2 above)
Comment (jkdouglas) [reply api-reference/beta/api/userflowattributes-delete.md:16]: Added this note

File: api-reference/beta/api/userflowattributes-get.md (hunk 3 above)
Comment (jkdouglas) [reply api-reference/beta/api/userflowattributes-get.md:85]: @odata.type is not returned in the API calls when testing them.

File: api-reference/beta/api/userflowattributes-update.md (hunk 4 above)
Comment (jkdouglas) [reply api-reference/beta/api/userflowattributes-update.md:16]: Added this note.

File: api-reference/beta/resources/userflowattributes.md (hunk 6 above)
Comment (jkdouglas) [reply api-reference/beta/resources/userflowattributes.md:29]: Added emphasis on the `custom user flow attribute` string.

File: api-reference/beta/resources/userflowattributes.md (hunk 5 above)
Comment (jkdouglas) [reply api-reference/beta/resources/userflowattributes.md:18]: Added link to this doc: https://docs.microsoft.com/azure/active-directory/develop/active-directory-schema-extensions

File: api-reference/beta/api/userflowattribute-post-userflowattributes.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+HTTP/1.1 201 Created
+Location: https://graph.microsoft.com/beta/identity/userFlowAttributes/extension_7a95ecd9489b4fb9a45722b913c4703b_Hobby
+Content-type: application/json
+
+{
+    "id": "extension_d09380e2b4c642b9a203fb816a04a7ad_Hobby",
+    "displayName": "Hobby",
+    "description": "Your hobby",
+    "userFlowAttributeType": "custom",
+    "dataType": "String"
```
Comment (John-YOU) [inline api-reference/beta/api/userflowattribute-post-userflowattributes.md:110]: 'String' should be 'string'

File: api-reference/beta/api/userflowattributes-list.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+{
+    "@odata.context": "https://graph.microsoft.com/beta/$metadata#userFlowAttributes",
+    "value": [
+      {
+          "id": "City",
+          "displayName": "City",
+          "description": "Your city",
+          "userFlowAttributeType": "builtIn",
+          "dataType": "String"
```
Comment (John-YOU) [inline api-reference/beta/api/userflowattributes-list.md:96]: Same here, 'String' should be all lowercase

File: api-reference/beta/resources/userflowattributes.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  "@odata.type": "microsoft.graph.identityUserFlowAttribute"
+} -->
+
+```json
+{
+    "id": "String (identifier)",
+    "displayName": "String",
+    "description": "String",
+    "userFlowAttributeType": "String",
+    "dataType": "String"
```
Comment (John-YOU) [inline api-reference/beta/resources/userflowattributes.md:55]: Same here

File: api-reference/beta/api/userflowattributes-get.md (hunk 3 above)
Comment (John-YOU) [reply api-reference/beta/api/userflowattributes-get.md:85]: During the implementation, I have difficulty adding it, so we didn't have the @odata.type returned in the API per previous discussion. Do you guys want me to have a fix to add it?

File: api-reference/beta/resources/userflowattributes.md (hunk 9 above)
Comment (jkdouglas) [reply api-reference/beta/resources/userflowattributes.md]: This is how the doc generator tool generates these and we need to follow that standard.

File: api-reference/beta/api/userflowattributes-list.md (hunk 8 above)
Comment (jkdouglas) [reply api-reference/beta/api/userflowattributes-list.md]: Docs standards have these as documented above and we need to follow that.

File: api-reference/beta/api/userflowattribute-post-userflowattributes.md (hunk 7 above)
Comment (jkdouglas) [reply api-reference/beta/api/userflowattribute-post-userflowattributes.md]: Doc standards document it as 'String' and we need to follow that

File: api-reference/beta/api/identityuserflowattribute-create.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+## Request body
+
+In the request body, provide a JSON representation of [identityUserFlowAttribute](../resources/identityuserflowattribute.md).
+
+|Property|Type|Description|
+|:---------------|:--------|:----------|
+|id|String|The ID of the user flow attribute. This is a read-only attribute that is automatically created.|
+|displayName|String|Required. The display name of the user flow attribute.|
+|description|String|Required. The description of the user flow attribute. It is shown to the user at the time of sign up.|
```
Comment (davidmu1) [inline api-reference/beta/api/identityuserflowattribute-create.md:56]:
```suggestion
|description|String|Required. The description of the user flow attribute. It is shown to the user at the time of sign-up.|
```

File: api-reference/beta/api/identityuserflowattribute-create.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+<!-- {
+  "blockType": "request",
+  "name": "create_userFlowAttribute_from_userFlowAttributes"
+}
+-->
+
+``` http
+POST https://graph.microsoft.com/beta/identity/userFlowAttributes
+Content-type: application/json
+Content-length: 154
```
Comment (davidmu1) [inline api-reference/beta/api/identityuserflowattribute-create.md:79]: Content-length isn't needed.

File: api-reference/beta/api/identityuserflowattribute-create.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+
+{
+  "displayName": "Hobby",
+  "description": "Your hobby",
+  "dataType": "string",
```
Comment (davidmu1) [inline api-reference/beta/api/identityuserflowattribute-create.md:84]: This should be one of the values listed in the definition for the property.

File: api-reference/beta/api/identityuserflowattribute-create.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+HTTP/1.1 201 Created
+Location: https://graph.microsoft.com/beta/identity/userFlowAttributes/extension_7a95ecd9489b4fb9a45722b913c4703b_Hobby
+Content-type: application/json
+
+{
+    "id": "extension_d09380e2b4c642b9a203fb816a04a7ad_Hobby",
+    "displayName": "Hobby",
+    "description": "Your hobby",
+    "userFlowAttributeType": "custom",
+    "dataType": "String"
```
Comment (davidmu1) [inline api-reference/beta/api/identityuserflowattribute-create.md:110]: Should this be "string"? The definition shows it as lowercase 's'.

File: api-reference/beta/api/identityuserflowattribute-create.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+
+Create a new [identityUserFlowAttribute](../resources/identityuserflowattribute.md) object.
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account)|IdentityUserFlow.ReadWrite.All|
```
Comment (davidmu1) [inline api-reference/beta/api/identityuserflowattribute-create.md:24]: Please make sure this permission is listed in the permissions-reference page.

File: api-reference/beta/api/identityuserflowattribute-get.md [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+
+Retrieve the properties and relationships of a [identityUserFlowAttribute](../resources/identityuserflowattribute.md).
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account)|IdentityUserFlow.Read.All, IdentityUserFlow.ReadWrite.All|
```
Comment (davidmu1) [inline api-reference/beta/api/identityuserflowattribute-get.md:24]: Please make sure these permissions are documented in the permissions-reference page.

File: api-reference/beta/api/identityuserflowattribute-get.md [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+
+{
+    "id": "City",
+    "displayName": "City",
+    "description": "Your city",
+    "userFlowAttributeType": "builtIn",
+    "dataType": "String"
```
Comment (davidmu1) [inline api-reference/beta/api/identityuserflowattribute-get.md:90]: lowercase?

File: api-reference/beta/api/identityuserflowattribute-update.md [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+<!-- {
+  "blockType": "request",
+  "name": "update_userFlowAttributes"
+}
+-->
+
+``` http
+PATCH https://graph.microsoft.com/beta/identity/userFlowAttributes/extension_d09380e2b4c642b9a203fb816a04a7ad_Hobby
+Content-type: application/json
+Content-length: 41
```
Comment (davidmu1) [inline api-reference/beta/api/identityuserflowattribute-update.md:77]: Content-length isn't needed here.

File: api-reference/beta/resources/enums.md [hunk 18]
```diff
@@ -1326,3 +1331,20 @@ Possible values for user account types (group membership), per Windows definitio
 | unknownFutureValue    | 2     | A sentinel member.                                                                                                                                                                                                 |
 
 
+### identityUserFlowAttributeDataType values
```
Comment (davidmu1) [inline api-reference/beta/resources/enums.md:1334]: Although it has been done in a few cases, values and descriptions shouldn't be defined in the enums.md file.

File: api-reference/beta/resources/identityuserflowattribute.md [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+author: "jkdouglas"
+---
+
+# identityUserFlowAttribute resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents user flow attributes in an Azure Active Directory tenant and an Azure AD B2C tenant.
```
Comment (davidmu1) [inline api-reference/beta/resources/identityuserflowattribute.md:16]:
```suggestion
Represents user flow attributes in an Azure Active Directory (Azure AD) tenant and an Azure AD B2C tenant.
```

File: api-reference/beta/resources/identityuserflowattribute.md [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+
+Configuring user flow attributes in your Azure Active Directory or Azure AD B2C tenant allows you collect information about a user during sign up. There is a built-in set of attributes that you can choose to collect, like Given Name, Surname, City, and Postal Code. You can also configure custom user flow attributes to collect information from a user that is not built-in to the directory. Custom user flow attributes are an abstraction over [Azure Active Directory schema extensions](https://docs.microsoft.com/azure/active-directory/develop/active-directory-schema-extensions).
```
Comment (davidmu1) [inline api-reference/beta/resources/identityuserflowattribute.md:18]:
```suggestion
Configuring user flow attributes in your Azure AD or Azure AD B2C tenant allows you to collect information about a user during sign-up. There is a built-in set of attributes that you can choose to collect. For example, Given Name, Surname, City, and Postal Code. You can also configure custom user flow attributes to collect information from a user that is not built in to the directory. Custom user flow attributes are an abstraction over [Azure Active Directory schema extensions](https://docs.microsoft.com/azure/active-directory/develop/active-directory-schema-extensions).
```

File: api-reference/beta/resources/identityuserflowattribute.md [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+
+## Methods
+
+| Method       | Return Type  |Description|
+|:---------------|:--------|:----------|
+|[List](../api/identityuserflowattribute-list.md)|identityUserFlowAttributes collection|Retrieve all built in and custom user flow attributes.|
```
Comment (davidmu1) [inline api-reference/beta/resources/identityuserflowattribute.md:24]:
```suggestion
|[List](../api/identityuserflowattribute-list.md)|identityUserFlowAttributes collection|Retrieve all built-in and custom user flow attributes.|
```

File: api-reference/beta/resources/identityuserflowattribute.md [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+|[Create](../api/identityuserflowattribute-create.md)|identityUserFlowAttribute|Create a new `custom user flow attribute`.|
```
Comment (davidmu1) [inline api-reference/beta/resources/identityuserflowattribute.md:25]:
```suggestion
|[Create](../api/identityuserflowattribute-create.md)|identityUserFlowAttribute|Create a new custom user flow attribute.|
```

File: api-reference/beta/resources/identityuserflowattribute.md [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+|[Get](../api/identityuserflowattribute-get.md) |identityUserFlowAttribute|Retrieve properties of a user flow attribute.|
+|[Update](../api/identityuserflowattribute-update.md)|None|Update a `custom user flow attribute`.|
```
Comment (davidmu1) [inline api-reference/beta/resources/identityuserflowattribute.md:27]:
```suggestion
|[Update](../api/identityuserflowattribute-update.md)|None|Update a custom user flow attribute.|
```

File: api-reference/beta/resources/identityuserflowattribute.md [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+|[Delete](../api/identityuserflowattribute-delete.md)|None|Delete a `custom user flow attribute`.|
```
Comment (davidmu1) [inline api-reference/beta/resources/identityuserflowattribute.md:28]:
```suggestion
|[Delete](../api/identityuserflowattribute-delete.md)|None|Delete a custom user flow attribute.|
```

File: api-reference/beta/resources/identityuserflowattribute.md [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+
+## Properties
+
+|Property|Type|Description|
+|:---------------|:--------|:----------|
+|id|String|The ID of the user flow attribute. This is a read-only attribute that is automatically created.|
```
Comment (davidmu1) [inline api-reference/beta/resources/identityuserflowattribute.md:34]:
```suggestion
|id|String|The identifier of the user flow attribute. This is a read-only attribute that is automatically created.|
```

File: api-reference/beta/resources/identityuserflowattribute.md [hunk 26]
```diff
@@ -0,0 +1,58 @@
...
+|displayName|String|The display name of the user flow attribute.|
+|description|String|The description of the user flow attribute. It is shown to the user at the time of sign up.|
```
Comment (davidmu1) [inline api-reference/beta/resources/identityuserflowattribute.md:36]:
```suggestion
|description|String|The description of the user flow attribute. It's shown to the user at the time of sign-up.|
```

File: api-reference/beta/toc.yml [hunk 27]
```diff
@@ -10299,6 +10299,20 @@ items:
...
+               href: api/identityuserflowattribute-create.md
+             - name: Get
+               href: api/identityuserflowattribute-get.md
+             - name: Update
+               href: api/identityuserflowattribute-update.md
+             - name: Delete
+               href: api/identityuserflowattribute-delete.md
       - name: Governance
         items:
         - name: Access reviews
```
Comment (davidmu1) [inline api-reference/beta/toc.yml:10318]: Don't forget your changelog entries.

File: api-reference/beta/api/identityuserflowattribute-create.md (hunk 13 above)
Comment (jkdouglas) [reply api-reference/beta/api/identityuserflowattribute-create.md:110]: Updated

File: api-reference/beta/api/identityuserflowattribute-get.md (hunk 16 above)
Comment (jkdouglas) [reply api-reference/beta/api/identityuserflowattribute-get.md:90]: Fixed

File: api-reference/beta/api/identityuserflowattribute-update.md (hunk 17 above)
Comment (jkdouglas) [reply api-reference/beta/api/identityuserflowattribute-update.md:77]: Removed

File: api-reference/beta/resources/identityuserflowattribute.md (hunk 23 above)
Comment (jkdouglas) [reply api-reference/beta/resources/identityuserflowattribute.md:27]: Fixed

File: api-reference/beta/resources/identityuserflowattribute.md (hunk 25 above)
Comment (jkdouglas) [reply api-reference/beta/resources/identityuserflowattribute.md:34]: Fixed

File: api-reference/beta/api/identityuserflowattribute-get.md (hunk 15 above)
Comment (jkdouglas) [reply api-reference/beta/api/identityuserflowattribute-get.md:24]: Added permissions reference for IdentityUserFlows

**Review by Lauragra — CHANGES_REQUESTED** [review]
Please add an update to changelog.md and the DirectoryServices.json changelog file. Otherwise, looks good!

File: api-reference/beta/api/identityuserflowattribute-update.md [hunk 28]
```diff
@@ -0,0 +1,94 @@
...
+
+## Request body
+
+In the request body, provide a JSON object with one or more properties that need to be updated for a [identityUserFlowAttribute](../resources/identityuserflowattribute.md) object.
+
+### userFlowAttributes object
+
+|Property|Type|Description|
+|:---------------|:--------|:----------|
+|description|String|The description of the user flow attribute. It is shown to the user at the time of sign up.|
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflowattribute-update.md:56]: Is description the only property that can be updated, or can they also update the display name? If the former, please adjust the sentence under request body. Either way, you don't need an H3 here, just say "provide a JSON object with the property/properties listed in the following table."

File: api-reference/beta/api/identityuserflowattribute-update.md [hunk 29]
```diff
@@ -0,0 +1,94 @@
...
+
+## Response
+
+If successful, this method returns a `204 No Content` response code. If unsuccessful, a `4xx` error will be returned with specific details.
+
+## Examples
+
+### Request
+
+The following is an example of the request.
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflowattribute-update.md:66]: Global: It's not a high priority, but FYI, in general we're looking at removing this boilerplate text in the Request/Response examples..

**Review by Lauragra — COMMENTED** [review]
Please clarify the properties that can be updated in the update topic. Otherwise looks good!

**Review by Lauragra — APPROVED** [review]

**Pull request merged**